CentOS/duffy
# Set up and initialize database

Fixes: #95 

This comes with setting up the `duffy` CLI to use sub-commands for different things: running the app, setting up the database, applying DB migrations, …

Additionally some generic changes:
- Rename virtualenv directory in GitHub actions
- Make tox runs less redundant
- Add duffy_config fixture and apply globally
- Make isort and black coexist well
- Fix import order

---------

How to test:

* Activate your virtual environment
* Run the test suite 😉
  ```
  tox
  ```

Manual tests:

* Copy the example configuration file…
  ```
  cp etc/duffy-example-config.yaml etc/duffy-config.yaml
  ```
* … and edit `database.sqlalchemy.*_url` to point to a real file, e.g. `devdata.db`:
  ```
  database:
    sqlalchemy:
      sync_url: "sqlite:///devdata.db"
      async_url: "sqlite+aiosqlite:///devdata.db"
  ```
* Set up the database, i.e. create tables from the DB model
  ```
  duffy -c etc/duffy-config.yaml setup-db
  ```
  The configured database file `devdata.db` should contain tables corresponding to what's defined in `duffy.database.model`, you can verify that in `sqlite3` with the command `.tables` or using the `sqliteman` GUI. 
* Run the app to serve the API:
  ```
  duffy --config etc/duffy-config.yaml serve --reload
  ```
  It should serve requests on http://127.0.0.1:8080
